modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java

```diff
@@ -448,26 +448,44 @@ public CodegenProperty fromProperty(String name, Schema p) {
             for (CodegenParameter p : op.allParams) {
                 p.vendorExtensions.put("x-go-example", constructExampleCode(p, modelMaps, processedModelMaps));
             }
+            processedModelMaps.clear();
         }
 
-        processedModelMaps.clear();
         for (CodegenOperation operation : operationList) {
+            boolean needTimeImport = false;
             for (CodegenParameter cp : operation.allParams) {
                 cp.vendorExtensions.put("x-go-example", constructExampleCode(cp, modelMaps, processedModelMaps));
+                if (cp.isDateTime || cp.isDate) { // datetime or date
+                    needTimeImport = true;
+                }
+            }
+            if (needTimeImport) {
+                operation.vendorExtensions.put("x-go-import", "    \"time\"");
             }
+            processedModelMaps.clear();
         }
 
         return objs;
     }
 
     private String constructExampleCode(CodegenParameter codegenParameter, HashMap<String, CodegenModel> modelMaps, HashMap<String, Integer> processedModelMap) {
         if (codegenParameter.isArray) { // array
-            return codegenParameter.dataType + "{" + constructExampleCode(codegenParameter.items, modelMaps, processedModelMap) + "}";
+            String prefix = codegenParameter.dataType;
+            String dataType = StringUtils.removeStart(codegenParameter.dataType, "[]");
+            if (modelMaps.containsKey(dataType)) {
+                prefix = "[]" + goImportAlias + "." + dataType;
+            }
+            return prefix + "{" + constructExampleCode(codegenParameter.items, modelMaps, processedModelMap) + "}";
         } else if (codegenParameter.isMap) {
-            return codegenParameter.dataType + "{ \"key\": " + constructExampleCode(codegenParameter.items, modelMaps, processedModelMap) + "}";
+            String prefix = codegenParameter.dataType;
+            String dataType = StringUtils.removeStart(codegenParameter.dataType, "map[string][]");
+            if (modelMaps.containsKey(dataType)) {
+                prefix = "map[string][]" + goImportAlias + "." + dataType;
+            }
+            return prefix + "{\"key\": " + constructExampleCode(codegenParameter.items, modelMaps, processedModelMap) + "}";
         } else if (codegenParameter.isPrimitiveType) { // primitive type
             if (codegenParameter.isString) {
-                if (StringUtils.isEmpty(codegenParameter.example)) {
+                if (!StringUtils.isEmpty(codegenParameter.example) && codegenParameter.example != "null") {
```

Comment:
@therve FYI. I've fled https://github.com/OpenAPITools/openapi-generator/pull/7998 to fix the string comparison in this line and a few other lines in this file.

Reply:
Sorry about. Not sure why I didn't catch that during testing :/

Reply:
No worries. No one complains. We spotted it with a code analyser.